types/uid-generator/index.d.ts

```diff
@@ -1,21 +1,22 @@
 // Type definitions for uid-generator 2.0
 // Project: https://github.com/nwoltman/node-uid-generator
 // Definitions by: TheEmrio <https://github.com/TheEmrio>
+//                 KyleChine <https://github.com/kylechine>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export interface UIDGenerator {
+interface UIDGeneratorInstance {
   readonly bitSize: number;
   readonly uidLength: number;
   readonly baseEncoding: string;
   readonly base: number;
   generateSync(): string;
   generate(): Promise<string>;
-  generate(cb: (uid: string) => any): void;
+  generate(cb: (error: Error|null, uid?: string) => any): void;
```

Comment:
```suggestion
  generate(cb: (error: Error|null, uid: string) => any): void;
```

Why did you put a `?` in front of the `uid` parameter?
When I look at the type definitions for `fs.readFile` for instance, I see
```ts
function readFile(path: PathLike | number, callback: (err: NodeJS.ErrnoException | null, data: Buffer) => void): void;
```
`data` here does not have a `?`. Maybe you should remove it here?

Reply:
From <https://github.com/nwoltman/node-uid-generator/blob/master/index.js>, Line 43 - 50:

```javascript
    crypto.randomBytes(this._byteSize, (err, buffer) => {
      if (err) {
        cb(err);
      } else {
        cb(null, bufferToString(buffer, this.baseEncoding, this.uidLength));
      }
    });
  }
```

When some error did triggered, the call-back function will be called as:

`cb(err);` which means the second parameter should be left as `undefined`

Reply:
I understand what you try to do, but I think that the convention in typescript is not to put a `?` in callback responses.
This is because the tsc server can't know that `err === null` means `uid` is a string.
For example, in `fs.readFile`, even tho `data` is null if there is an error, the type still doesn't have a `?` 
Additionally, it would be a hassle always to write `uid!` instead of `uid` x)
Typescript types assume that developers know that they should check the `err` parameter before using callback values.